src/libs/actions/IOU.ts

```diff
@@ -798,7 +798,6 @@ function buildOnyxDataForMoneyRequest(
                 // Disabling this line since transaction.filename can be an empty string
                 // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
                 errors: getReceiptError(transaction.receipt, transaction.filename || transaction.receipt?.filename, isScanRequest, errorKey),
-                pendingAction: null,
```

Comment:
We shouldn't clear the pending action when fails, otherwise, we don't have the information on whether the action is new or not. That's the pattern we use, for example, when adding a comment/message.

Additional video showing the transaction is deleted when it's a new transaction.
https://github.com/Expensify/App/assets/50919443/c9fa65ce-ae59-47f3-9c8d-401ce3375130